corehq/apps/sms/views.py

```diff
@@ -2151,8 +2151,15 @@ def page_context(self):
             )
         else:
             wa_active_backend = turn_backend.get() if turn_backend.count() else infobip_backend.get()
-            templates = wa_active_backend.get_all_templates()
-            for template in templates:
-                template['template_string'] = wa_active_backend.generate_template_string(template)
-            context.update({'wa_templates': templates})
+            try:
+                templates = wa_active_backend.get_all_templates()
+                for template in templates:
+                    template['template_string'] = wa_active_backend.generate_template_string(template)
+                context.update({'wa_templates': templates})
+            except Exception:
```

Comment:
Would be preferable to catch something more specific than `Exception` because this will silence bugs (example: `Exception` would catch a fat-finger coding mistake that causes the try to always raise an exception like `AttributeError`). Is it possible to change it to a more specific exception?

Reply:
Yes, good point. I've converted it to a more specific - _TypeError_ (since we are trying to catch 'NoneType' object is not iterable exception here). Alternatively I can get rid of try-except and handle it by adding one IF statement.

Reply:
> Alternatively I can get rid of try-except and handle it by adding one IF statement.

This is an excellent idea. Let's do that.

Reply:
Done.